rudder-core/src/main/scala/com/normation/rudder/services/policies/DataStructures.scala

```diff
@@ -299,16 +314,9 @@ final case class Policy(
   , overrides          : Set[(RuleId,DirectiveId)] //a set of other draft overriden by that one
 ) extends Loggable {
 
-  def toUnboundBoundedPolicyDraft() = ExpandedUnboundBoundedPolicyDraft(
-    technique         = technique
-  , directiveId       = id.directiveId
-  , priority          = priority
-  , isSystem          = isSystem
-  , policyMode        = policyMode
-  , trackerVariable   = trackerVariable
-  , variables         = expandedVars
-  , originalVariables = originalVars
-  )
+  val expandedVars    = Policy.mergeVars(policyVars.map( _.expandedVars.values).toList.flatten)
```

Comment:
there is a change of behaviour here: policy were ordered by priority, and if equal, alphanumerically
it had an impact on the order of variable, and usage of uniqueVariable.

this is replaced here by an unordered set, for both expandedVars and originalVars, which may result in different order for both, and it breaks completely the uniqueVariable usage - unless its done somewhere else, and I failed to see it (and in this case, we don't need to use lists)

Reply:
my bad, after talking with @fanf , he explained me that policyVars are sorted, and the map().toList kept the order

Reply:
It is still sorted, but I will add comments to make it clearer.